app/src/main/java/org/mozilla/fenix/perf/NavGraphProvider.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import androidx.lifecycle.LifecycleCoroutineScope
+import androidx.navigation.NavController
+import java.util.WeakHashMap
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.R
+
+/**
+ * This class asynchronously loads the Navigation xml.
```

Comment:
Can we explain why we asynchronously load the navigation XML? i.e. for performance benefits

It's convey to say why a class exists so if, at some point, other devs can determine if it's no longer necessary and remove it.